sdk/core/core/inc/az_static_assert.h

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_STATIC_ASSERT_H
+#define AZ_STATIC_ASSERT_H
+
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AZ_STATIC_ASSERT(CONDITION) typedef int az_static_assert[(CONDITION) ? 1 : -1];
```

Comment:
We really need some comments in these files. I don't know why or how someone would use this macro.
Also, do we really need this macro? Each thing we add, we MUST test, document, and support FOREVER! So, we want the surface area to be really tiny. The bar to add something like this is really, really high. There must be significant value. When in doubt, leave it out.